src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -967,6 +981,20 @@ static grpc_lb_policy *glb_create(grpc_exec_ctx *exec_ctx,
   glb_policy->args = grpc_channel_args_copy_and_add_and_remove(
       args->args, args_to_remove, GPR_ARRAY_SIZE(args_to_remove), &new_arg, 1);
 
+  /* Fallback is off */
+  glb_policy->fallback = false;
+
+  /* Copy the backend server addresses from the resolver for fallback */
+  grpc_lb_addresses *backends = grpc_lb_addresses_create(
+      addresses->num_addresses - num_grpclb_addrs, NULL);
+  for (size_t addr_copied = 0, i = 0; i < addresses->num_addresses; ++i) {
```

Comment:
I chose to use another loop because:

- I need `num_grpclb_addrs` before copying backends, and `num_grpclb_addrs` is a result of the existing for loop. There is a dependency. 
- These two loops have different goals. The old one is for validation; the new one is for data movement. 

(Maybe moving the new loop to under the old one can help compiler optimization like loop fusion?)